client/jetpack-cloud/index.js

```diff
@@ -26,15 +27,22 @@ const clearPageTitle = ( context, next ) => {
 	next();
 };
 
-const redirectToPrimarySiteLanding = ( context ) => {
+const redirectToPrimarySiteLanding = ( context, next ) => {
 	debug( 'controller: redirectToPrimarySiteLanding', context );
 	const state = context.store.getState();
 	const currentUser = getCurrentUser( state );
 	const isPrimarySiteJetpackSite = getPrimarySiteIsJetpack( state );
 
-	isPrimarySiteJetpackSite
-		? page( `/landing/${ currentUser.primarySiteSlug }` )
-		: page( `/landing` );
+	const isAgencyDashboardEnabled = config.isEnabled( 'jetpack/agency-dashboard' );
+
+	if ( isAgencyDashboardEnabled ) {
+		page( `/dashboard` );
+	} else {
+		isPrimarySiteJetpackSite
+			? page( `/landing/${ currentUser.primarySiteSlug }` )
+			: page( `/landing` );
+	}
+	next();
```

Comment:
What is the purpose of calling `next()` since every code path leads the user to somewhere via a call to `page(...)`?